### PR TITLE
Fix "Invalid array length" bug with spread of set values

### PR DESCRIPTION
**Pull request by efritz**
This should fix the issue reported in https://sourcegraph.atlassian.net/jira/servicedesk/projects/SG/queues/custom/1/SG-267.

### CODE REVIEW

File: shared/search/providers.ts [hunk 1]
```diff
@@ -527,9 +527,9 @@ function sortByProximity(locations: sourcegraph.Location[], currentURI: URL): so
 function jaccardIndex<T>(a: Set<T>, b: Set<T>): number {
     return (
         // Get the size of the intersection
-        new Set([...[...a]].filter(value => b.has(value))).size /
```
Comment (twop) [inline shared/search/providers.ts:530]: Now I'm really curios why it was the way it was and what made it go  away? Tanspiling?

Comment (efritz) [reply shared/search/providers.ts:530]: https://github.com/sourcegraph/code-intel-extensions/pull/384/files#diff-d7d6c42b22b4cc3c9e90d696d64e26eeR530 🤷

Comment (felixfbecker) [reply shared/search/providers.ts:530]: This is really interesting because while the code before was already double-copying the arrays (not good), the autofixed double spread looks equivalent and not like it shouldn't work or cause the RangeError 🤔

File: shared/tsconfig.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
         "tag": "gql",
       },
     ],
+    "downlevelIteration": true,
```
Comment (felixfbecker) [inline shared/tsconfig.json:12]: Why is this needed? We only target modern browsers so generally shouldn't do this as it could bloat bundles. Is this significant to fix the bug?

Comment (efritz) [reply shared/tsconfig.json:12]: This is the only thing that fixes the bug. The double spread is just a drive-by.

File: shared/search/providers.ts (hunk 1 above)
Comment (efritz) [reply shared/search/providers.ts:530]: This didn't fix the error, I just cleaned this up since I noticed it. I have no clue why it was doing a double copy in the first place.
